File: cpp/Wilk.cpp
#include "Wilk.h"
#include "Komentarz.h"

Wilk::Wilk(Swiat *swiat, Punkt pole, int turaPowstania)
    : Zwierze(swiat, pole, turaPowstania, sila, inicjatywa, symbol){
    this->symbol = SYMBOL_WILKA;
    this->turaPowstania = turaPowstania;
    this->sila = SILA_WILKA;
    this->inicjatywa = INICJATYWA_WILKA;
    this->pole = pole;
    this->swiat = swiat;
    this->umarl = false;
}

string Wilk::nazwaOrganizmu() {
    return "Wilk";
}

File: java/pl/edu/pg/po/projekt2/zwierzeta/Wilk.java
package pl.edu.pg.po.projekt2.zwierzeta;

import pl.edu.pg.po.projekt2.Punkt;
import pl.edu.pg.po.projekt2.Swiat;
import pl.edu.pg.po.projekt2.Zwierze;

public class Wilk extends Zwierze {
    private static final char symbol = 'W';
    private static final int inicjatywa = 5;

    public Wilk(Swiat swiat, Punkt pole, int turaPowstania) {
        super(swiat, pole, turaPowstania, 9, inicjatywa, symbol);
        this.sila = 9;
        this.pole = pole;
        this.swiat = swiat;
        this.umarl = false;
        this.turaPowstania = turaPowstania;
    }

    @Override
    public String nazwaOrganizmu() {
        return "Wilk";
    }
}
